pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java

```diff
@@ -214,20 +217,34 @@ public void addOrReplaceOfflineSegment(String tableName, String segmentId, boole
     }
   }
 
-  private boolean isNewSegmentMetadata(SegmentMetadata segmentMetadataFromServer,
-      SegmentMetadata segmentMetadataForCheck, String segmentName, String tableName) {
-    if (segmentMetadataFromServer == null || segmentMetadataForCheck == null) {
-      LOGGER.info("segmentMetadataForCheck = null? {}, segmentMetadataFromServer = null? {} for {} of table {}",
-          segmentMetadataForCheck == null, segmentMetadataFromServer == null, segmentName, tableName);
+  private boolean isNewSegmentMetadata(@Nonnull OfflineSegmentZKMetadata newSegmentZKMetadata,
+      @Nullable SegmentMetadata existedSegmentMetadata) {
+    String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(newSegmentZKMetadata.getTableName());
+    String segmentName = newSegmentZKMetadata.getSegmentName();
+
+    if (existedSegmentMetadata == null) {
+      LOGGER.info("Existed segment metadata is null for segment: {} in table: {}", segmentName, offlineTableName);
       return true;
     }
-    LOGGER.info("segmentMetadataForCheck.crc={},segmentMetadataFromServer.crc={} for {} of table {}",
-        segmentMetadataForCheck.getCrc(), segmentMetadataFromServer.getCrc(), segmentName, tableName);
-    if ((!segmentMetadataFromServer.getCrc().equalsIgnoreCase("null"))
-        && (segmentMetadataFromServer.getCrc().equals(segmentMetadataForCheck.getCrc()))) {
-      return false;
+
+    long newCrc = newSegmentZKMetadata.getCrc();
```

Comment:
I think it's good to leave `TODO` or a comment to indicate that you try to keep the same logic as before. I can see we are not checking creation time here while the controller side checks it. Eventually, this part can even be removed as we discussed offline.